File: OrderNet-Server/src/main/java/bitc/fullstack503/ordernetserver/dto/app/OrderAppDTO.java
package bitc.fullstack503.ordernetserver.dto.app;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.Data;
import java.math.BigDecimal;
import java.util.Date;

@Data
public class OrderAppDTO {

  @JsonProperty("order_id")
  private String orderId;

  @JsonProperty("order_date")
  private Date orderDate;

  @JsonProperty("order_due_date")
  private Date orderDueDate;

  @JsonProperty("branch_id")
  private String branchId;

  @JsonProperty("branch_name")
  private String branchName;

  @JsonProperty("part_id")
  private String partId;

  @JsonProperty("part_name")
  private String partName;

  @JsonProperty("part_cate")
  private String partCate;

  @JsonProperty("order_item_quantity")
  private int orderItemQuantity;

  @JsonProperty("order_item_price")
  private BigDecimal orderItemPrice;

  @JsonProperty("order_price")
  private BigDecimal orderPrice;

  @JsonProperty("order_item_status")
  private String orderItemStatus;

  @JsonProperty("total_quantity")
  private int totalQuantity;  // 총 품목 수량

  @JsonProperty("order_status")
  private String orderStatus; // 주문 상태
}

File: OrderNet-APP/app/src/main/java/bitc/fullstack/app/dto/OrderAppDTO.kt
package bitc.fullstack.app.dto

import com.google.gson.annotations.SerializedName
import java.math.BigDecimal
import java.util.Date

data class OrderAppDTO(
    @SerializedName("order_id")
    var orderId: String,

    @SerializedName("order_date")
    var orderDate: Date,

    @SerializedName("order_due_date")
    var orderDueDate: Date,

    @SerializedName("branch_id")
    var branchId: String,

    @SerializedName("branch_name")
    var branchName: String,

    @SerializedName("part_id")
    var partId: String,

    @SerializedName("part_name")
    var partName: String,

    @SerializedName("part_cate")
    var partCate: String,

    @SerializedName("order_item_quantity")
    var orderItemQuantity: Int,

    @SerializedName("order_item_price")
    var orderItemPrice: BigDecimal,

    @SerializedName("order_price")
    var orderPrice: BigDecimal,

    @SerializedName("order_item_status")
    var orderItemStatus: String,

    @SerializedName("warehouse_stock")
    var warehouseStock: Int?,

    @SerializedName("total_quantity")
    var totalQuantity: Int,

    @SerializedName("order_status")
    var orderStatus: String,

    @SerializedName("warehouse_order_price")
    var warehouseOrderPrice: BigDecimal
)
